### DISCUSSION TITLE
GraphQL: X does not have the correct permissions to execute `AddProjectV2ItemById`

### DISCUSSION THREAD

**Discussion by alopix** [community/community#40838] [Apps, API and Webhooks] [answered]
### Select Topic Area

Question

### Body

I am not sure if this is just a question or an actual bug.

Trying to use GraphQL and adding items to the new GitHub Projects sometimes returns the error `X does not have the correct permissions to execute `AddProjectV2ItemById`.
The personal access token has `project` read and write permissions so not sure why this error would show up. Also, it does not seem to be consistent?!

The same can happen with `UpdateProjectV2ItemFieldValue`.

**Comment by ossanna16** [comment]
Hi there @alopix and welcome to our community! Thank you for asking a great question 🙂
While you are waiting for a reply to your question, here are a few things you can do:

To get started, introduce yourself in our [official introduction thread](https://github.com/orgs/community/discussions/23204)
To connect with other new users, check out this Discussion to [Meet Your GitHub Community Team](https://github.com/orgs/community/discussions/28776), further your GitHub knowledge by working through our [GitHub Skill 🆙 exercises](https://github.com/orgs/community/discussions/32121), and participate in our [weekly Fun Friday chats](https://github.com/orgs/community/discussions/30912)

**Comment by alopix** [comment, marked as answer]
I figured out the answer myself. The error is not related to the personal access token system but happens when you have not given the user running the query write access to the (modern) GitHub project.

**Comment by rmusick2000** [comment]
I had the same error using explorer (https://developer.github.com/v4/explorer/), but in this case it was a token issue.  

More specifically, the mutation addProjectV2ItemById returned the  "INSUFFICIENT_SCOPES" error: "The 'addProjectV2ItemById' field requires one of the following scopes: ['project'], but your token has only been granted ...".

In this case the user had full access to the repo and the project.  It seems the explorer does not use the personal access token owned by the user, but some generic one that does not have write permissions.  Switching to javascript and explicitly sending the user's (only) PAT, addProjectV2ItemById worked as expected.
